### ISSUE TITLE
bug: RDS Instance and Cluster Deletion Fails (Timing Issue) 

### ISSUE THREAD

**Issue by hoffmc** [localstack/localstack#7601]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When destroying an RDS cluster, which contains an RDS instance, the cluster deletion will fail on the first attempt:

```
aws_rds_cluster_instance.db-project: Still destroying... [id=database-project-localstack, 10s elapsed]
aws_rds_cluster_instance.db-project: Still destroying... [id=database-project-localstack, 20s elapsed]
aws_rds_cluster_instance.db-project: Still destroying... [id=database-project-localstack, 30s elapsed]
aws_rds_cluster_instance.db-project: Destruction complete after 30s
aws_rds_cluster.postgres-cluster-project: Destroying... [id=localstack-postgres-cluster]
╷
│ Error: deleting RDS Cluster (localstack-postgres-cluster): InvalidDBClusterStateFault: Cluster cannot be deleted, it still contains DB instances in non-deleting state.
│       status code: 400, request id: 09049Z6QNRV02K0K5NRC83LT761WRR6XLMPRO0374IWVZB80RIEC
```

Performing the same destroy command again, it will succeed, even though the cluster does not exist any more. This indicates that localstack returned 400, but the cluster was finally deleted.:
```
Do you really want to destroy all resources?
  Terraform will destroy all your managed infrastructure, as shown above.
  There is no undo. Only 'yes' will be accepted to confirm.

  Enter a value: yes

aws_security_group.local: Destroying... [id=sg-6bee7014f35077550]
aws_rds_cluster_parameter_group.aurora-cluster-parameter-group: Destroying... [id=local-db-cluster]
aws_rds_cluster_parameter_group.aurora-cluster-parameter-group: Destruction complete after 0s
aws_security_group.local: Destruction complete after 0s
aws_vpc.main: Destroying... [id=vpc-f7aa6159]
aws_vpc.main: Destruction complete after 0s
```

### Expected Behavior

Cluster destroy should not return an error.

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run localstack/localstack-pro:1.3.1

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

We experienced this issue both with terraform and pulumi:

* terraform 1.3.7 with provider plugin 4.50.0
* pulumi 3.53.1 with pulumi-aws provider 5.28.0

terraform sample code:
```
...
terraform {
  required_providers {
    aws = {
      version = "~> 4.50.0"
    }
  }
}
...
resource "aws_rds_cluster" "postgres-cluster" {
  availability_zones                  = ["eu-west-1a"]
  cluster_identifier                  = "localstack-postgres-cluster"
  engine                              = "aurora-postgresql"
  engine_mode                         = "provisioned"
  db_cluster_parameter_group_name     = aws_rds_cluster_parameter_group.aurora-cluster-parameter-group.name
  iam_database_authentication_enabled = false
  master_username                     = "postgres"
  master_password                     = "postgres"
  database_name                       = "postgres"
  port                                = "4510"
  vpc_security_group_ids              = [aws_security_group.local.id]
  deletion_protection                 = true
  skip_final_snapshot                 = true
}

resource "aws_rds_cluster_instance" "db-instance" {
  availability_zone  = "eu-west-1a"
  cluster_identifier = aws_rds_cluster.postgres-cluster.id
  engine             = aws_rds_cluster.postgres-cluster.engine
  engine_version     = "13.4"
  identifier         = "database-localstack"
  instance_class     = "db.t4g.micro"
}

resource "aws_rds_cluster_parameter_group" "aurora-cluster-parameter-group" {
  name        = "local-db-cluster"
  family      = "aurora-postgresql13"
  description = "Parameter Group for Aurora Cluster"

  parameter {
    name         = "shared_preload_libraries"
    value        = "pg_stat_statements,pgaudit"
    apply_method = "pending-reboot"
  }

  parameter {
    name         = "pgaudit.role"
    value        = "rds_pgaudit"
    apply_method = "pending-reboot"
  }

  parameter {
    name         = "pgaudit.log"
    value        = "All"
    apply_method = "pending-reboot"
  }

  parameter {
    name         = "log_connections"
    value        = 1
    apply_method = "pending-reboot"
  }

  parameter {
    name         = "log_disconnections"
    value        = 1
    apply_method = "pending-reboot"
  }
}

resource "aws_security_group" "local" {
  name        = "local"
  description = "localstack"
  vpc_id      = aws_vpc.main.id
}
```

### Environment

```markdown
- OS: Ubuntu 20.04 LTS, MacOS 13
- LocalStack: 1.3.1 Pro
```

### Anything else?

The problem only occurs when using Localstack. No problem with AWS. It occurs both with terraform and pulumi, on a Mac development environment and on an Ubuntu based pipeline.

**Comment by steffyP** [comment]
Hi @hoffmc,

thanks for reporting and attaching a reproducible example!
The issue is fixed in the `localstack-pro:latest` image. 
Could you please pull the `latest` image and confirm that the use case is now working for you as expected?

**Comment by hoffmc** [comment]
Hi @steffyP ,

The fix works, thanks! Will it be included with the upcoming 1.4 release?

**Comment by steffyP** [comment]
Glad to hear! Yes, it will be part of the 1.4 release.
